VultisigApp/VultisigApp/Services/CoinService/EvmCoinService.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  EvmCoinService.swift
+//  VultisigApp
+//
+//  Created by Enrique Souza Soares on 05/07/24.
+//
+
+import Foundation
+
+@MainActor
+class EvmCoinService: CoinService {
```

Comment:
Can we do not use inheritance? We can do it with protocols

Reply:
What about the code duplication?

Reply:
or you can always use CoinService instance. add there addDiscoveredTokens and addTokenToChain

Reply:
Evms have different implementations. That is why I am using polymorphism.